contracts/exchange/test/exchange_transfer_simulator_test.ts

```diff
@@ -95,7 +88,7 @@ describe('ExchangeTransferSimulator', async () => {
             const balanceAndProxyAllowanceLazyStore = new BalanceAndProxyAllowanceLazyStore(
                 simpleERC20BalanceAndProxyAllowanceFetcher,
             );
-            exchangeTransferSimulator = new ExchangeTransferSimulator(balanceAndProxyAllowanceLazyStore);
+            exchangeTransferSimulator = new ExchangeTransferSimulator(balanceAndProxyAllowanceLazyStore, env.provider);
```

Comment:
What do you think of passing an instance of `DevUtilsContract` instead of the provider? It's what I did in `LocalBalanceStore`. 

Each instance of `DevUtilsContract` instantiates its own EVM and keeps it around in memory so having a lot of instances around might start to affect performance. I also talked to @BMillman19 about making it a convention to instantiate contract wrappers once per app wherever possible.